libappstream-glib/as-store.c

```diff
@@ -264,6 +265,23 @@ as_store_changed_uninhibit_cb (void *v)
 
 #define _cleanup_uninhibit_ __attribute__ ((cleanup(as_store_changed_uninhibit_cb)))
 
+static GPtrArray *
+_dup_app_array (GPtrArray *array)
+{
+	GPtrArray *array_dup;
+
+	if (array == NULL)
+		return g_ptr_array_new_with_free_func ((GDestroyNotify) g_object_unref);
```

Comment:
Do we need these two lines? Does g_ptr_array_new_full explode in a ball of fire if you try to allocate it a size 0?

Reply:
Setting it to 0 should be fine. We could do something like:

```
GPtrArray *array_dup = g_ptr_array_new_full (array != NULL ? array->len : 0, (GDestroyNotify) g_object_unref);`

if (array == NULL)
        return g_steal_pointer (&array_dup);

...
```

Would that be cleaner?

Reply:
Ohh! `array == NULL`, i read that as as `array->len == 0` for some reason. What you had was cleaner, ignore me. I do wonder if callers to this function should check for NULL tho....

Reply:
I think it's easier to just return an empty container here because in all the public API that did "return g_object_ref (priv->array)" so far (and that I've changed to return a deep copy now) it was always returning an empty array and not NULL and clients probably rely on that now.

Reply:
Right, but wouldn't a `g_return_val_if_fail(array !=NULL)` be a sensible precaution instead?